txcache/selectionTracker.go

```diff
@@ -103,16 +103,8 @@ func (st *selectionTracker) OnProposedBlock(
 		return err
 	}
 
-	st.blocks[string(blockHash)] = tBlock
-
-	if len(st.blocks) == int(st.maxTrackedBlocks) {
-		log.Warn("selectionTracker.OnProposedBlock: max tracked blocks reached",
-			"len(st.blocks)", len(st.blocks),
-		)
-
-		// clear all the proposed blocks
-		st.blocks = make(map[string]*trackedBlock)
-	}
+	// adds the new block in the map of tracked blocks, overwriting the block with the same nonce
```

Comment:
Maybe this comment can be dropped, since you have documented `addNewTrackedBlockNoLock`?